db/migrate/20221202082725_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,6 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts,
+                  :user, null: false, foreign_key: true
+  end
+end
```

Comment:
- [ ] It is always easier to create a reference immediately when you are creating the table for posts instead of again coming to add it here. An example of it fone together would look something like this:

```
create_table :posts do |t|
    t.string :title
    t.text :text
    t.integer :comments_counter
    t.integer :likes_counter
    t.timestamps # this will crate uopdated_at and created_at columns
    t.references :author, foreign_key: { to_table: 'users' } # this will create author_id and a corresponding index  (but students can also add column and index in separate commands)
  end
```

Also, kindly note that in your ERD diagram, there is supposed to be a author_id and not user_id as you have here. To do it your way, this can simply be done this way:

```
def change
    add_reference :posts, :author, null: false, foreign_key: {to_table: :users}
  end
```

This will give you the same outcome as the "doing it together in one place" I have shown you above.

Reply:
Thanks alot

Reply:
This change has been done well